middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -619,6 +619,22 @@ let close_primitive acc env ~let_bound_var named (prim : Lambda.primitive) ~args
     in
     (* Since raising of an exception doesn't terminate, we don't call [k]. *)
     Expr_with_acc.create_apply_cont acc apply_cont
+  | (Pmakeblock _ | Pmakefloatblock _ | Pmakearray _), [] ->
+    (* Special case for liftable empty blocks *)
+    begin
+      match prim with
+      | Pmakeblock (tag, _, _) when tag <> 0 ->
+        (* There should not be any way to reach this from Ocaml code. *)
+        Misc.fatal_error
+          "Non-zero tag on empty block allocation in [Closure_conversion]"
+      | _ -> ()
```

Comment:
Please make this an exhaustive match (we should enable warning 4 everywhere at some point...)